### PR TITLE
Refine Latex Tables with Group Cols

### PR DESCRIPTION
**Pull request by kbrevoort**
# Summary

I was making some tables for a paper I'm writing this afternoon and found a few minor issues when Latex tables are created with row_group_as_column = TRUE that this PR addresses.  There are three changes:

1. The multirow Latex command used for the group column use an "=" which instructs Latex to respect the column widths that have been selected.  However, when no column widths have been selected, it defaults to using all the width available.  This PR changes this behavior, replacing the "=" with "*", so the natural width of the column will be used.
2. There was a bug in the code that caused an error when assigning widths to the group column that I've fixed.  (There was a boolean vector where I wanted to count the TRUE values, but instead was taking the length of the vector.)
3. Stubhead labels were not being properly centered when widths were specified.  The stubhead label is now properly centered.

This Quarto code:

```
---
title: "Untitled"
execute: 
  echo: false
format: pdf
---

```{r}
#| include: false
library(tidyverse)
library(gt)
```

```{r}
tbl <- select(exibble, group, row, num, currency) |>
  gt(rowname_col = 'row', 
     groupname_col = 'group',
     row_group_as_column = TRUE) |>
  tab_row_group(label = 'Group 1', 
                rows = 1:3) |>
  tab_stubhead('Stubhead') |>
  tab_options(latex.use_longtable = TRUE)

tbl
```
currently produces this output:

Screenshot 2024-07-18 at 1 35 04 AM

With this PR, the same code will now produce:

Screenshot 2024-07-18 at 1 36 14 AM

# Related GitHub Issues and PRs

None.  

# Checklist

- [x ] I understand and agree to the [Code of Conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html).
- [x ] I have listed any major changes in the [NEWS](https://github.com/rstudio/gt/blob/master/NEWS.md).
- [x ] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/rstudio/gt/tree/master/tests/testthat) for any new functionality.

### CODE REVIEW

File: R/utils_render_latex.R [hunk 1]
```diff
@@ -405,23 +405,17 @@ create_columns_component_l <- function(data, colwidth_df) {
...
-        if (stub_df$pt > 0) {
-          sprintf("%.2fpt", stub_df$pt)
-        } else if (stub_df$lw > 0) {
-          sprintf("%.2f\\linewidth", stub_df$lw)
-        } else {
-          ""
-        }
-
-      if (stub_width == "") {
+      stub_df <- dplyr::filter(colwidth_df, type %in% c('stub', 'row_group'))
```
Comment (olivroy) [inline R/utils_render_latex.R:408]:
```suggestion
      stub_df <- dplyr::filter(colwidth_df, type %in% c("stub", "row_group"))
```

**Review by olivroy — APPROVED** [review]
Thanks for your diligence! Please add news for the change! 

I think that this fixes #1799.

**Comment by olivroy** [comment]
@kbrevoort Could you run `source("scripts/visual-test-latex.R")` and commit the result?

**Comment by olivroy** [comment]
@kbrevoort Wow! It actually fixes #1799. See the second table in the visual examples. The second page of https://github.com/rstudio/gt/blob/8744851023d7bee722d74bfd9693b17cb3b232e2/pkgdown/assets/gt-latex.pdf no longer is baffled 

Currently: 
image

**Comment by rich-iannone** [comment]
Thank you @kbrevoort and @olivroy ! Will merge this in now.

**Pull request merged**